### PR TITLE
Add images test

### LINKED ISSUE #184: Add tests for images

### PR DESCRIPTION
**Pull request by takufunkai**
Fixes #184

### CODE REVIEW

File: src/test/java/seedu/address/logic/commands/DeleteImageCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    @Test
+    public void constructor_negativeIndex_throwsIndexOutOfBoundsException() {
+        assertThrows(IndexOutOfBoundsException.class, () -> new DeleteImageCommand(Index.fromZeroBased(-1)));
+    }
+
+    @Test
+    public void execute_success() {
+        Person personToRemoveImage = model.getFilteredPersonList().get(INDEX_SECOND_PERSON.getZeroBased());
+        Index imageToDeleteIndex = Index.fromOneBased(1);
```
Comment (weijuey) [inline src/test/java/seedu/address/logic/commands/DeleteImageCommandTest.java:46]: Should this be `FromZeroBased`? Is it ok to just use `INDEX_SECOND_PERSON` here?

File: src/test/java/seedu/address/logic/commands/DeleteImageCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        CommandResult expectedResult = new CommandResult(String.format(
+                DeleteImageCommand.MESSAGE_DELETE_IMAGE_SUCCESSFUL, imageToDeleteIndex.getOneBased(), editedPerson),
+                VIEW_IMAGES);
+        expectedModel.setPerson(personToRemoveImage, editedPerson);
+
+        assertCommandSuccess(deleteImageCommand, model, expectedResult, expectedModel);
+    }
+
+    @Test
+    public void execute_nullIndex_throwsNullPointerException() {
```
Comment (weijuey) [inline src/test/java/seedu/address/logic/commands/DeleteImageCommandTest.java:61]: Might want to clarify which index is null here, as you did in the rest of the test method names! :smiley:

File: src/test/java/seedu/address/logic/commands/ImagesCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        populateTestImages();
+        UserPrefs userPrefs = new UserPrefs();
+        userPrefs.setContactImagesFilePath(TEST_IMAGES_DIRECTORY);
+
+        model = new ModelManager(getTypicalAddressBook(), userPrefs);
+        expectedModel = new ModelManager(model.getAddressBook(), userPrefs);
+    }
+
+    @Test
+    public void execute_nullIndex_throwsNullPointerException() {
```
Comment (weijuey) [inline src/test/java/seedu/address/logic/commands/ImagesCommandTest.java:39]: Same as in DeleteCommandTest

**Review by weijuey — CHANGES_REQUESTED** [review]
Great job on making very detailed tests, as well as creating useful utility classes just for testing Ezekiel :+1: . Lgtm, just some small nits!

File: src/main/java/seedu/address/logic/commands/ImagesCommand.java [hunk 4]
```diff
@@ -63,28 +64,31 @@ public ImagesCommand() {
...
         }
 
         Person targetPerson = lastShownList.get(index.getZeroBased());
-        logger.info(String.format("Sanitizing images of person at index %d", index.getZeroBased()));
+        logger.info(String.format("Sanitizing images of person at index %d, at file path: %s", index.getZeroBased(),
+                model.getContactImagesFilePath()));
 
         ImageDetailsList originalList = targetPerson.getImageDetailsList();
-        ImageDetailsList sanitizedList = ImageUtil.sanitizeList(originalList);
+        ImageDetailsList sanitizedList = ImageUtil.sanitizeList(originalList, model.getContactImagesFilePath());
```
[the remark this answers is not in the document]
Comment (takufunkai) [reply src/main/java/seedu/address/logic/commands/ImagesCommand.java:80]: Great catch, totally missed that out. Thanks a lot!

File: src/test/java/seedu/address/logic/commands/CommandTestUtil.java [hunk 5]
```diff
@@ -114,6 +114,50 @@ public static void assertCommandFailure(Command command, Model actualModel, Stri
...
+        // we are unable to defensively copy the model for comparison later, so we can
+        // only do so by copying its components.
+        AddressBook expectedAddressBook = new AddressBook(actualModel.getAddressBook());
+        List<Person> expectedFilteredList = new ArrayList<>(actualModel.getFilteredPersonList());
+
+        assertThrows(CommandException.class, expectedMessage, () -> command.executeInDetailedView(actualModel));
+        assertEquals(expectedAddressBook, actualModel.getAddressBook());
+        assertEquals(expectedFilteredList, actualModel.getFilteredPersonList());
+    }
+
```
[the remark this answers is not in the document]
Comment (takufunkai) [reply src/test/java/seedu/address/logic/commands/CommandTestUtil.java:160]: No worries! Your component is a lot of burden to carry, I'm hoping I can help you as much as possible!

File: src/test/java/seedu/address/logic/commands/DeleteImageCommandTest.java (hunk 1 above)
Comment (takufunkai) [reply src/test/java/seedu/address/logic/commands/DeleteImageCommandTest.java:46]: I felt that the name `INDEX_FIRST_PERSON` does not correspond to the index of the image (while they are the same "value", the name suggests different things)

I'm trying to get the first (zero-th) image from the `personToRemoveImage`. So it should be `Index.fromOneBased(1)` or, `Index.fromZeroBased(0)`. 

Do you think it's good to make the change to `INDEX_FIRST_PERSON`?

Comment (weijuey) [reply src/test/java/seedu/address/logic/commands/DeleteImageCommandTest.java:46]: Sorry, my bad, I misread this part and didn't realise that this was the image index, and wasn't related to the line above 😰. I think using `1` is ok.

**Review by weijuey — APPROVED** [review]

**Pull request merged**